Added machine line

doekman opened the pull request:
I've added a "Machine" line. My Pascal is rusty, and I've never really used Asm in my code: any tips?

Code from: [DOS_SDK](https://github.com/Scalibq/DOS_SDK/blob/main/ASM/8259A.asm)

doekman committed:
Added machine line

leahneukirchen commented:
I'm not very good at it either 😅 and don't even have a means to test this...

doekman commented:
😅 Let me try to find some people with actual hardware to test this. Fingers crossed 🤞

xTibor commented on dosfetch.pas:
```diff
@@ -138,6 +138,88 @@ procedure fpu;
+        pop es
+        mov a, cl
+    end;
+    { Couldn't get this to work with Case-statement in combination with Const-values }
```
Make the constants typeless ordinals (drop the `byte` type specifier) and you can match them in a case statement.

doekman replied:
This has to wait a couple of weeks, since I'm away from my desktop.

doekman committed:
Removed type from const

This way hexadecimal values can be used, and integer values can be used in the case-statement

doekman commented:
The only XT about the "IBM XT Model 286" is they used the XT case. The machine detection checks if there is an additional interrupt controller, and concludes if there are two, it will be an AT. 

See:

* https://www.homecomputermuseum.nl/en/collectie/ibm/ibm-5162/
* https://www.dosdays.co.uk/computers/IBM%20PC-XT-286%20(5162)/ibm5162.php

So machine names are not very exact. Not sure what to do about it. Options:

* Leave it like this. IBM XT Model 286 is more AT than XT (and there where only made 20k of them)
* Don't implement this option. 
  - Better try to detect CPU
  - Or show number of interrupt controllers/PIC's and bus type

However, I probably won't implement much more. I'll happy withdraw this PR if it's not the direction you want this tool to go.

polluks commented:
What about the PCjr?
https://www.vogons.org/viewtopic.php?t=50417

ltning commented:
Would it be a point to limit the scope of this to detecting the CPU family? 8086/88, V[23]0/186, 286, 386(sx,dx), 486 and perhaps a single final "586+"? :)
